bibtex-actions-latex.el

```diff
@@ -0,0 +1,118 @@
+      "citeauthor*" "Citeauthor*" "citetitle" "citetitle*" "citeyear"
+      "citeyear*" "citedate" "citedate*" "citeurl" "fullcite"
+      "footfullcite" "notecite" "Notecite" "pnotecite" "Pnotecite"
+      "fnotecite") . (["Prenote"] ["Postnote"] t))
+    (("nocite" "supercite") . nil))
+  "Citation commands and their argument specs. The argument spec is the same
+as the args argument of `TeX-parse-macro'. When calling
+`bibtex-actions-insert-citation' the keys will be inserted at the position
+where `TeX-parse-macro' leaves the point."
+  :group 'bibtex-actions-latex
+  :type '(alist :key-type (repeat string)
+                :value-type sexp))
+
+(defcustom bibtex-actions-latex-prompt-for-extra-arguments t
+  "Whether to prompt for additional arguments when inserting a citation."
+  :group 'bibtex-actions-latex
+  :type 'boolean)
+
+(add-to-list 'bibtex-actions-major-mode-functions
+             '((latex-mode) .
+               ((local-bib-files . bibtex-actions-latex--local-bib-files)
+                (insert-keys . bibtex-actions-latex--insert-keys)
+                (insert-citations . bibtex-actions-latex--insert-citations)
+                (keys-at-point . bibtex-actions-latex--keys-at-point))))
+
+(defun bibtex-actions-latex--local-bib-files ()
+  "Retrieve local bibliographic files for a latex buffer using reftex."
+  (reftex-locate-bibliography-files (TeX-master-directory)))
+
+(defun bibtex-actions-latex--keys-at-point ()
+  "Returns a list of keys at point in a latex buffer."
+    (when (bibtex-actions-latex-is-a-cite-command (TeX-current-macro))
+      (split-string (thing-at-point 'list t) "," t "[{} ]+")))
+
+(defun bibtex-actions-latex--insert-keys (keys)
+  "Insert comma sperated KEYS in a latex buffer."
+  (insert (string-join keys ", ")))
+
+(defvar bibtex-actions-latex-cite-command-history nil
+  "Variable for history of cite commands")
+
+(defun bibtex-actions-latex--insert-citations (keys &optional command)
+  "Insert a citation consisting of KEYS.
+
+If the command is inside a citation command keys are added to it. Otherwise
+a new command is started.
+If the optional COMMAND is provided use it, otherwise prompt for one.
+The availiable commands and how to provide them arguments are configured
+by `bibtex-actions-latex-cite-commands'. If
+`bibtex-actions-latex-prompt-for-extra-arguments' is `nil`, every
+command is assumed to have a single argument into which keys are inserted."
+  (if (bibtex-actions-latex-is-a-cite-command (TeX-current-macro))
+      (progn (skip-chars-forward "^,}")
+             (unless (equal ?} (preceding-char)) (insert ", ")))
+    (let ((macro (or command
+                     (completing-read "Cite command: "
+                                    (seq-mapcat #'car bibtex-actions-latex-cite-commands)
+                                    nil nil nil
+                                    'bibtex-actions-latex-cite-command-history nil nil))))
+      (TeX-parse-macro macro
```

Comment:
This works very well except I am getting a `cite: info not available, use ‘C-c &’ to get it.` message I think sue to `TeX-parse-macro`. I need to figure out why and how to suppress it.

Reply:
Seems to have resolved itself.